### PR TITLE
Ex1 son

### PR DESCRIPTION
**Pull request by tranngocson210599**

### CODE REVIEW

File: Queue/Queue.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+{
+	q.data = new int[size];
+	q.capacity = size;
+	q.front = 0;
+	q.rear = -1;
+	q.count = 0;
+}
+
+
+int size(Queue& q)
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:14]: Corrent the naming convention. Should be GetSize or something like that.

File: Queue/Queue.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+{
+	return q.count;
+}
+
+
+
+bool isFull(Queue& q)
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:21]: First character of function should be Upper case.

File: Queue/Queue.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+	q.count++;
+}
+
+
+int DeQueue(Queue& q)
+{
+	if (isEmpty(q))
+	{
+		cout << "Queue is Empty";
+		return 0;
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:49]: We should return -1 for errror, and 0 for success. 0 is a general success return code for function.

File: Queue/Queue.cpp [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	q.front = (q.front + 1) % q.capacity;
+	q.count--;
+}
+int Peek(Queue& q)
+{
+	if (isEmpty(q))
+	{
+		cout << "Queue is Empty" << endl;
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:58]: Same comments as above. We need something to know if the Peek function is called successful or not by using a return code negative (for example -1)

File: Queue/Queue.cpp [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	{
+		int index = (q.front + i) % q.capacity;
+		cout << q.data[index] << endl;
+	}
+
+}
+//selection sort
+void Sort(Queue& q)
+{
+	int i, j, min_idx;
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:74]: We should return immediately if the size is 0 or 1.

File: Queue/Queue.cpp [hunk 6]
```diff
@@ -0,0 +1,86 @@
+#include"Queue.h"
+using namespace std;
+
+void CreateQueue(Queue& q, int size)
+{
+	q.data = new int[size];
```
Comment (nguyenvangiang1988) [inline Queue/Queue.cpp:6]: I don't see you free the data after ending your program. Can you run Valgrind to check if we are leaking the memory or not?

File: Queue/Queue.cpp (hunk 1 above)
Comment (tranngocson210599) [reply Queue/Queue.cpp:14]: I done it already

File: Queue/Queue.cpp (hunk 2 above)
Comment (tranngocson210599) [reply Queue/Queue.cpp:21]: I have changed it

File: Queue/Queue.cpp (hunk 3 above)
Comment (tranngocson210599) [reply Queue/Queue.cpp:49]: I done

File: Queue/Queue.cpp (hunk 4 above)
Comment (tranngocson210599) [reply Queue/Queue.cpp:58]: I done it

File: Queue/Queue.cpp (hunk 5 above)
Comment (tranngocson210599) [reply Queue/Queue.cpp:74]: I solved it, you can review it and check if it is optimized

**Pull request merged**